components/clinical/plan-definition-editor/src/index.tsx

```diff
@@ -0,0 +1,240 @@
+const StyledListItem = styled.li`
+  margin-bottom: 1rem;
+
+  ${WIDESCREEN_MINIMUM_MEDIA_QUERY} {
+  }
+`
+
+const StyledSectionToggle = styled.div`
+  background-color: ${BANNER_COLOURS.INFO.BACKGROUND};
+  font-weight: bold;
+  color: black;
+  display: flex;
+  align-items: center;
+  padding: 0.5rem 0.25rem;
+  ${toggleInputMixin}
+`
+
+const StyledBullseyeIcon = styled(BullseyeIcon)``
+
+const StyledProblemSection = styled.ul`
+  list-style: none;
+  padding: 0.25rem 0.25rem;
+  margin-bottom: 1rem;
+`
+
+export const SelectedPlanDetail: FC<ISelectedPlanDetailProps> = ({
+  planDefinition,
+  selectedProblemIds,
+  onProblemChange,
+}) => {
+  // Split actions from plan definition into clinical problem based and education based
+  const { problemActions, educationActions } = useMemo(() => {
+    const pa = planDefinition.action?.filter(
+      (action) =>
+        action?.code && action?.code[0] && action?.code[0].coding && action?.code[0].coding[0]?.code === 'problem'
+    )
+    const ea = planDefinition.action?.filter(
+      (action) =>
+        action?.code && action?.code[0] && action?.code[0].coding && action?.code[0].coding[0]?.code === 'education'
+    )
+
+    return {
+      problemActions: pa,
+      educationActions: ea,
+    }
+  }, [planDefinition.action])
+
+  // Calculate whether all problems in the given categories are enabled
+  const { allProblemsOn, allEducationOn } = useMemo(
+    () => ({
+      allProblemsOn:
+        problemActions?.map((pa) => pa?.elementId as string).every((id) => selectedProblemIds.includes(id)) ?? false,
+      allEducationOn:
+        educationActions?.map((ea) => ea?.elementId as string).every((id) => selectedProblemIds.includes(id)) ?? false,
+    }),
+    [educationActions, problemActions, selectedProblemIds]
+  )
+
+  const handleSingleActionChange: OnProblemChange = (e) => {
+    onProblemChange(e)
+  }
+
+  const handleToggleAllProblemsClick = (e: ChangeEvent<HTMLInputElement>) => {
+    const newState = problemActions?.map((p) => ({ problemId: p?.elementId as string, state: e.target.checked }))
+    newState && onProblemChange(newState)
+  }
+
+  const handleToggleAllEducationClick = (e: ChangeEvent<HTMLInputElement>) => {
+    const newState = educationActions?.map((p) => ({ problemId: p?.elementId as string, state: e.target.checked }))
+
+    newState && onProblemChange(newState)
+  }
+
+  return (
+    <>
+      <StyledSectionToggle>
+        <div>
+          <ToggleInput
+            checked={allProblemsOn}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => handleToggleAllProblemsClick(e)}
+            id="problem-toggle-all"
+          />
+        </div>
+        <label htmlFor="problem-toggle-all">Patient Problems</label>
+      </StyledSectionToggle>
+      <StyledProblemSection>
+        {problemActions?.map((problem, pidx) => {
+          const goal = planDefinition.goal?.find(
+            (g) => problem?.goalId && g?.elementId?.includes(problem.goalId[0] as string)

```

Comment:
Is it worth adding a `problem?.goalId.length > 0` check in here as well as the truthiness?

Reply:
Defo.

Reply:
☝️ "Professional"